File: 프로그래머스/lv0/120878. 유한소수 판별하기/유한소수 판별하기.py
def gcd(n, m):
    if m == 0:
        return n
    return gcd(m, n%m) if n>m else gcd(n, m%n)

def solution(a, b):
    b = b // gcd(a,b)
    
    i = 2
    data = []
    while i <= b:
        if b%i == 0:
            b = b // i
            if i not in data:
                data.append(i)
        else:
            i += 1
    print(data)
    if data == [] or data == [5,2] or data == [2,5] or data == [2] or data == [5]:
        return 1
    return 2

File: 프로그래머스/0/120878. 유한소수 판별하기/유한소수 판별하기.java
class Solution {
    public int solution(int a, int b) {
        b /= gcd(a, b);
        
        while (b % 2 == 0) {
            b /= 2;
        }
        
        while (b % 5 == 0) {
            b /= 5;
        }
        
        return b == 1 ? 1 : 2;
    }

    public int gcd(int a, int b) {
        if (b == 0) {
            return a;
        }
        return a > b ? gcd(b, a % b) : gcd(a, b % a);
    }
}